### PR TITLE
fix getting route method, and fix constructer implements

### PR DESCRIPTION
**Pull request by ophum**
Please excuse my poor English.

### fix route method in `routing/routing.go`
route method always return default gateway when routing table contained default gateway.
I think default gateway if `rt.Src` and `rt.Dst` is nil.
return default gateway at last if static route is not found.
so I fixed it in [L132-L152](https://github.com/ophum/gopacket/blob/a0bf7fd45b3d65fd7c494b87eec6d57ef2cb9ef8/routing/routing.go#L132-L152)

### fix constructor implements
creating `rtr.ifaces` of `map[int]*net.Interface`  in New method. but assigned value is pointer of `for` statement 's variable.
then contents of `rtr.ifaces` is all same pointer.
so I fixed it in [L226-L227](https://github.com/ophum/gopacket/blob/a0bf7fd45b3d65fd7c494b87eec6d57ef2cb9ef8/routing/routing.go#L226-L227)

### CODE REVIEW

**Comment by ophum** [comment]
@googlebot I signed it!

**Comment by gconnell** [comment]
This looks great!  would you potentially be able to add a test for it?

**Comment by ophum** [comment]
@gconnell thanks!

I tried writing tests. 

`TestPrivateRoute` is  test of `router.route()` method.
`TestRouting` is test of Router.Route method (this test is internally using New, router.RouteWithSrc, router.route).

## `TestRouting`
use test environment built by netns . 
this test must run by root user because using netns. so append below command  to  `.travis.script.sh`.
```
sudo $(which go) test github.com/google/gopacket/routing
```

tests run in child netns . 
topo

### SetUp
create virtual environment by below process 
1. create parent netns ([routing_test.go#L287-L290](https://github.com/google/gopacket/blob/873c589dbba98a4eef0f287a927afa5953bfb6f9/routing/routing_test.go#L287-L290))
1. create child netns ([routing_test.go#L291-L294](https://github.com/google/gopacket/blob/873c589dbba98a4eef0f287a927afa5953bfb6f9/routing/routing_test.go#L291-L294))
1. create `veth0`, `veth1` (peer name is suffixed `-peer`) ([routing_test.go#L295-L320](https://github.com/google/gopacket/blob/873c589dbba98a4eef0f287a927afa5953bfb6f9/routing/routing_test.go#L295-L320))
1. set ip address  `192.168.10.1/24` to `veth0`, `192.168.20.1/24` to `veth1` ([routing_test.go#L321-L342](https://github.com/google/gopacket/blob/873c589dbba98a4eef0f287a927afa5953bfb6f9/routing/routing_test.go#L321-L342))
1. set up `veth0` and `veth1`. ([routing_test.go#L343-L354](https://github.com/google/gopacket/blob/873c589dbba98a4eef0f287a927afa5953bfb6f9/routing/routing_test.go#L343-L354))
1. set `veth0-peer` and `veth1-peer` to parent netns and up its. ([routing_test.go#L355-L386](https://github.com/google/gopacket/blob/873c589dbba98a4eef0f287a927afa5953bfb6f9/routing/routing_test.go#L355-L386))

routing table is below at this time.
```
192.168.10.0/24 dev veth0 proto kernel scope link src 192.168.10.1
192.168.20.0/24 dev veth1 proto kernel scope link src 192.168.20.1
```

### Testing

Test below cases.
- exists route in static route without default gateway ([routing_test.go#L393-L419](https://github.com/google/gopacket/blob/873c589dbba98a4eef0f287a927afa5953bfb6f9/routing/routing_test.go#L393-L419))
- not exists route in static route without default gateway ([routing_test.go#L420-L434](https://github.com/google/gopacket/blob/873c589dbba98a4eef0f287a927afa5953bfb6f9/routing/routing_test.go#L420-L434))
- exists route in static route with default gateway ([routing_test.go#L435-L474](https://github.com/google/gopacket/blob/873c589dbba98a4eef0f287a927afa5953bfb6f9/routing/routing_test.go#L435-L474))
- not exists route in static route with default gateway ([routing_test.go#L475-L513](https://github.com/google/gopacket/blob/873c589dbba98a4eef0f287a927afa5953bfb6f9/routing/routing_test.go#L475-L513))

**Comment by ophum** [comment]
merge branch master into this branch, and fix conflict (both accept...).

File: routing/routing.go [hunk 1]
```diff
@@ -214,7 +223,8 @@ loop:
 	if err != nil {
 		return nil, err
 	}
-	for _, iface := range ifaces {
+	for _, tmp := range ifaces {
```
Comment (njuptlzf) [inline routing/routing.go:226]: NIT: Can we keep it as it is?

File: routing/routing.go [hunk 2]
```diff
@@ -214,7 +223,8 @@ loop:
...
+		iface := tmp
```
Comment (njuptlzf) [inline routing/routing.go:227]: NIT: Can we keep it as it is?

Comment (gconnell) [reply routing/routing.go:227]: This is a correct bug-fix as-is, since future code takes an address of iface.

**Comment by gconnell** [comment]
THanks again a ton, and appologies for forgetting about this.

**Comment by ophum** [comment]
thanks!

**Pull request merged**